C-Plus-Plus/ds/Vector.cpp

```diff
@@ -0,0 +1,210 @@
+in it, then a new array of twice the size is declared and all the elements from the
+previous array are copied onto the new array. This is an expensive operation
+
+push_back() Time Complexity: O(current size)
+
+*/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+template<typename T>
+class Vector {
+
+	int cs; // Current Size : Gives count of the elements currently in vector
+	int ms; // Maximum Size : Gives count of the maximum elements that can be in vector
+	T *arr;	// Array declared
+public:
+	// Constructor
+	Vector() {
+		cs = 0;
+		ms = 1;
+		// Array is dynamically allocated memory space
+		arr = new T[ms];
+	}
+
+	// Inserts new element into the vector from back
+	void push_back(const T d) {
+
+		if (cs == ms) {
+			// Array is full
+			T *oldArr = arr;
+			// arr points to a new memory location with twice memory space
+			arr = new T[2 * ms];
+			ms = 2 * ms;
+			// Copy elements from oldArr
+			for (int i = 0; i < cs; i++) {
+				arr[i] = oldArr[i];
+			}
+			// clear the old memory
+			delete [] oldArr;
+		}
+
+		// Insert new element
+		arr[cs] = d;
+		cs++;
+	}
+
+	// Removes last element from the vector
+	void pop_back() {
+		cs--;
+	}
+
+	// Returns the element at the front of the vector
+	T front() const {
+		return arr[0];
+	}
+
+	// Returns the element at the back of the vector
+	T back() const {
+		return arr[cs - 1];
+	}
+
+	// Returns whether vector is empty
+	bool empty() const {
+		return cs == 0;
+	}
+
+	// Returns current size of the vector
+	int size() const {
+		return cs;
+	}
+
+	// Returns maximum size of the vector
+	int capacity() const {
+		return ms;
+	}
+
+	// Returns iterator to the start of the vector
+	T* begin() const {
+		return arr;
+	}
+
+	// Returns iterator to the end of the vector
+	T* end() const {
+		return arr + cs;
+	}
+
+	// Returns element at ith index
+	T operator[](const int i) {
+		return arr[i];
+	}
+
+	// Clears the vector
+	void clear() {
+		cs = 0;
+	}
+};
+
+void printVector(Vector<int> v) {
+	for (int i = 0; i < v.size(); i++) {
+		cout << v[i] << " ";
+	} cout << endl;
+}
+
+int main() {
+
+	cout << "Enter <1> To push elements \n"
+	     "<2> To pop (last) element \n"
+	     "<3> To get front element \n"
+	     "<4> To get back element\n"
+	     "<5> To check if vector is empty \n"
+	     "<6> To get size of vector \n"
+	     "<7> To get capacity of vector \n"
+	     "<8> To get iterator to start of vector \n"
+	     "<9> To get iterator to the end of vector \n"
+	     "<10> To clear vector \n"
+	     "<11> To print the vector \n"
+	     "<12> To exit program \n\n";
+
+	cout << "Enter Choice: ";
+
+	// Declaring a vector
+	Vector<int> v;
+
+	int choice;
+
+	while (cin >> choice) {
+
+		switch (choice) {
+
+		case 1: cout << "Enter elements to push until -1: ";
+			int x;
+			cin >> x;
+			while (x != -1) {
+				v.push_back(x);
+				cin >> x;
+			} break;
+
+		case 2: if(!v.empty()) cout << v.pop_back() << endl; 
```

Comment:
```suggestion
		case 2: if(!v.empty()) v.pop_back(); 
```
v.pop_back() function just pops out the last element in the vector, it is a void function so it won't return any value to be printed using cout.

Reply:
Yeah. It is so embarrassing